### PR TITLE
Lmb 557 |  email on effectief older than 10 days

### PR DESCRIPTION
**Pull request by JonasVanHoof**
## Description

An email must be send to the bestuurseeneheid contact when there is a mandataris with publication status effectief for 10 or more days.

## How to test

Run the deliver mail service togheter with this PR and see if eny mails are created in the mandataris logs. Than look at the app deliver-email service logs to see if the mail is created and you can see a preview url.

## Links to other PR's

- https://github.com/lblod/app-lokaal-mandatenbeheer/pull/218
- https://github.com/lblod/frontend-lokaal-mandatenbeheer/pull/328

## Attachments

image

### CODE REVIEW

File: util/create-email.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  PREFIX nie: <http://www.semanticdesktop.org/ontologies/2007/01/19/nie#>
+  PREFIX nfo: <http://www.semanticdesktop.org/ontologies/2007/03/22/nfo#>
+
+  INSERT DATA {
+    GRAPH <http://mu.semte.ch/graphs/system/email> {
+      <http://data.lblod.info/id/emails/${uuid()}> a nmo:Email;
+        nmo:messageFrom ${from};
+        nmo:emailTo ${to};
+        nmo:messageSubject "Besluit voor mandataris";
+        nmo:plainTextMessageContent "Mandataris met uri ${mandatarisUri} staat al 10 of meer dagen op publicatie status effectief zonder dat er een besluit aan hangt.";
```
Comment (JonasVanHoof) [inline util/create-email.ts:23]: What do we want to say here and doe sthis need to be html?

File: util/create-email.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import { v4 as uuid } from 'uuid';
+import { sparqlEscapeDateTime, sparqlEscapeString } from './mu';
+import { updateSudo } from '@lblod/mu-auth-sudo';
+
+const EMAIL_FROM_MANDATARIS_EFFECTIEF =
+  process.env.EMAIL_FROM_MANDATARIS_EFFECTIEF ??
+  'lokaal-mandatenbeheer@vlaanderen.be';
```
Comment (JonasVanHoof) [inline util/create-email.ts:7]: Fictieve email

File: cron/notification-for-bekrachtigde-mandataris.ts [hunk 3]
```diff
@@ -27,19 +29,24 @@ async function HandleEffectieveMandatarissen() {
...
+            uri: mandatarisWithGraph.mandataris.value,
           },
         ],
       });
+      const email = await bestuurseenheid_sudo.getContactEmailFromMandataris(
+        mandatarisWithGraph.mandataris,
+      );
+      if (email) {
+        await sendMailTo(email.value, mandatarisWithGraph.mandataris.value);
+      }
```
Comment (JonasVanHoof) [inline cron/notification-for-bekrachtigde-mandataris.ts:49]: What if there is no email set?

File: cron/notification-for-bekrachtigde-mandataris.ts [hunk 4]
```diff
@@ -5,6 +5,8 @@ import { getSparqlResults } from '../util/sparql-result';
 import { MANDATARIS_STATUS } from '../util/constants';
 import { sparqlEscapeDateTime } from '../util/mu';
 import { createNotification } from '../util/create-notification';
+import { bestuurseenheid_sudo } from '../data-access/bestuurseenheid';
+import { sendMailTo } from '../util/create-email';
 
 const NOTIFICATION_CRON_PATTERN =
```
Comment (JonasVanHoof) [inline cron/notification-for-bekrachtigde-mandataris.ts:11]: DO we need a separate cron time for emails?
DO we need to be able to toggle emails on and off? (docker-compose)

**Comment by JonasVanHoof** [comment]
Perfect !
image

File: util/create-email.ts [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+}
+console.log(
+  `EMAIL_FROM_MANDATARIS_EFFECTIEF SET TO: ${EMAIL_FROM_MANDATARIS_EFFECTIEF}`,
+);
+console.log(`SEND_EMAIL_FOR_MANDATARIS_EFFECTIEF SET TO: ${SEND_EMAILS}`);
+
+export async function sendMailTo(emailTo: string, mandataris) {
+  const from = sparqlEscapeString(EMAIL_FROM_MANDATARIS_EFFECTIEF as string);
+  const to = sparqlEscapeString(emailTo);
+  const description = `De publicatie status van ${mandataris.name} met mandaat ${mandataris.mandate} en uri ${mandataris.uri} staat al 10 dagen of meer op effectief zonder dat er een besluit is toegevoegd. Gelieve deze mandataris manueel te bekrachtigen en een besluit toe te voegen of publiceer het besluit van de installatievergadering via een notuleringspakket.`;
```
Comment (bfidlers) [inline util/create-email.ts:21]: Improved the mail a little bit, but could still be better, but I would keep it like this, until Tim asks otherwise.

**Review by bfidlers — APPROVED** [review]
Looks good!

**Pull request merged**